### PR TITLE
Drink quality affects mood

### PR DESCRIPTION
**Pull request by Mickyan**
:cl: Mickyan
add: Mixed drinks now give mood boosts with varying strength depending on their quality.
add: Although powerful, mood boosts from quality drinks are short lived. If you want to make the most out of them, take a sip every few minutes like a normal human being instead of downing the entire glass like the alcoholic you are.
/:cl:
Time to get fancy, bartenders. The harder it is to make a drink, the stronger the effect, usually.
There are exceptions, but quality tiers look something like this:
1. any mixed drink
2. 3+ ingredients, at least one alcoholic and non-alcoholic each
3. either takes some particular effort to make to make or requires exotic ingredients
4. requires exotic ingredients and take some effort to make

I also slightly lowered the strenght of the drunk moodlet to encourage using mixed drinks, though getting drunk out of regular drinks is still quite effective due to how long lasting it is.

I haven't played bartender in a while so if some of the values look off to you let me know

### CODE REVIEW

**Comment by Qustinnus** [comment]
Being overly drunk should probably be a bad mood.

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 1]
```diff
@@ -153,6 +153,7 @@ All effects don't start immediately, but rather get worse over time; the rate is
 	color = "#102000" // rgb: 16, 32, 0
 	nutriment_factor = 1 * REAGENTS_METABOLISM
 	boozepwr = 80
+	quality = 2
```
Comment (Qustinnus) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: use defines preferably

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 2]
```diff
@@ -13,12 +13,27 @@
...
 
 /datum/reagent/consumable/on_mob_life(mob/living/carbon/M)
 	current_cycle++
 	M.nutrition += nutriment_factor
 	holder.remove_reagent(src.id, metabolization_rate)
 
+/datum/reagent/consumable/reaction_mob(mob/living/M, method=TOUCH, reac_volume)
+	if(method == INGEST)
+		if (quality && !M.has_trait(TRAIT_AGEUSIA))
+			switch(quality)
```
Comment (Qustinnus) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: because then this becomes more sensical if you have 4 defines

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by Qustinnus** [comment]
also; you forgot to update the DME

File: code/modules/reagents/chemistry/reagents/food_reagents.dm (hunk 2 above)
Comment (Mickyan) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: makes sense, should I put them here on top of food_reagents or with the other food defines?

**Comment by Mickyan** [comment]
bad mood for drinking too much should probably come from it's effects (vomiting, confusion, passing out etc) rather than being drunk in itself

**Comment by Kagus** [comment]
Is there anything in place to prevent a person from mixing the four different levels of these together and making a supercocktail that provides +10 mood? And, furthermore, do we want such a thing in place?

Also, as an aside, chemistry can make Fringe Weaver (currently rated at +3) by the gallon. It's not exactly a tricky mix, although service doesn't really have a great source of ethanol handy under normal circumstances... So I guess that just makes chemistry more popular for picking up some space drug-laced fringe weaver?

Might want to bump lizard wine up to fantastic considering the scarcity of lizard tails (which then also need ethanol, which again comes from chemistry... Hmm...) so it's actually a desirable luxury item rather than being on par with a host of other, more easily attainable drinks. Depends though, since its manner of creation means that buffing it wouldn't really be a buff for service/bartending in general.

No love for Bacchus' Blessing? It doesn't have a drink quality assigned to it at all. Same for Squirt Cider, but considering the flavor description... Yeah, can't really blame ya for not giving that one a happy moodlet. Also botany can just grow plants that naturally produce the stuff, so yarr.

Oh, shit... That's right... Fruit wines. I guess just leaving it at "good" and telling its shifting flavors and alcohol contents to piss off is really just good enough, it's not like anyone uses fermentation barrels anyways.

Any thoughts regarding the fernet cocktails, since fernet is now technically contraband and you need a hacked booze-o-mat in order to dispense it? While that does officially increase the difficulty of making those cocktails, it doesn't change the fact that fernet and coke tastes like mouthwash.

**Comment by Qustinnus** [comment]
Mood code only allows one moodlet of each type. This means you'd get the +4 moodlet only.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm (hunk 2 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: You can put them on either depending on how localized you want it to be.

**Comment by ExcessiveUseOfCobblestone** [comment]
Is there any way to make this a singular type and it maxes the mood to the best quality drink (resets timer or whatever) rather than 4 different moodlets you can potentially stack, or did you intend for them to be stackable?

**Comment by Qustinnus** [comment]
you cant stack them. the drinks are all the same category of moodlet.

**Comment by ExcessiveUseOfCobblestone** [comment]
So then being drunk is objectively better than casually sipping your drink since it's also in the same category? Speak slowly please I have special needs.

**Comment by Qustinnus** [comment]
`SEND_SIGNAL(src, COMSIG_ADD_MOOD_EVENT, "drunk", /datum/mood_event/drinks/drunk)`
drunk is the category

```
		if (DRINK_NICE)
					SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "quality_drink", /datum/mood_event/drinks/quality_nice)
				if (DRINK_GOOD)
					SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "quality_drink", /datum/mood_event/drinks/quality_good)
				if (DRINK_VERYGOOD)
					SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "quality_drink", /datum/mood_event/drinks/quality_verygood)
				if (DRINK_FANTASTIC)
					SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "quality_drink", /datum/mood_event/drinks/quality_fantastic)
```
quality_drink is the category

**Comment by ExcessiveUseOfCobblestone** [comment]
If only you could make your code as easy on the eyes as you are

**Comment by Mickyan** [comment]
Yeah I should probably lower fringe weaver and bump up lizard wine

Ones such as Bacchus Blessing, Hooch and Moonshine are by their flavor text and appearance, really shitty drinks that just make you really drunk so their lack of quality reflects that regardless of the difficulty in making it

**Comment by Kagus** [comment]
I somehow doubt the tastiness factor of such classics as "acid spit" and "neurotoxin", and I'm not sure if changeling sting's ability to make you taste your own brain coming out of your nose is meant as a positive, but alright... I can at least see that logic. It just makes me a little sad that the booziest booze in the game (also found in a Lavaland ruin!) doesn't get any desirable qualities that can validate it getting made.

@Qustinnus Ah, there it is. Yeah, just skimmed a bit, didn't see that they were subtypes like that.

**Comment by Qustinnus** [comment]
its not the subtypes its the way you call the proc

**Comment by Kagus** [comment]
It is late and I've been debugging my alcohol cabinet, they're subtypes of ways to call the proc. So there.

**Comment by Mickyan** [comment]
it'll still count like you mixed correctly

**Comment by octareenroon91** [comment]
>it'll still count like you mixed correctly

Strong, weak, and poisoned drinks still taste nice.

**Review by SpaceManiac — APPROVED** [review]
Code looks fine, works in testing, can be rebalanced later if needed.

**Pull request merged**